### ISSUE TITLE
feat(docs): add localization

### ISSUE THREAD

**Issue by MFarabi619** [cuhacking/2025#26]
**What feature should we improve? Please describe.**
Localization is the process of adapting content to support different languages and cultural contexts. Currently, our docs are only in English, which limits accessibility for non-English speakers. Localizing our content demonstrates our commitment to accessibility and inclusivity for the hackathon.

**Describe the solution you'd like**
I would like to start by adding French localization to the documentation site. This includes integrating a localization features within the Fumadocs framework, providing a language switcher, and ensuring all content and UI elements are translated into French. The feature should auto-detect what the user's system preference is for language, and use the corresponding language on initial load.

**Describe alternatives you've considered**
I've considered using automatic translation services like Google Translate, but this may not provide accurate or culturally relevant translations. I've also considered localizing only key sections, but this could lead to inconsistencies.

**Additional context**
Implementing localization will require collaboration with native French speakers in our team to ensure accurate translations.
Visibility test suites also need to re-architected to account for localized content.

**Comment by HasithDeAlwis** [comment]
Side note for this, we should consider a solution that will also work with our entire site, accessibility for all languages is important. I've worked with `react-i18n` not sure on other solutions, so we need to explore more. `react-i18n` does auto detect what language a users system is configured to and we can easily provide a language switcher using it.

**Comment by MFarabi619** [comment]
Yessir it's already on the roadmap. The rest of the app and the landing page will have localization as well.

**Comment by JeremyFriesenGitHub** [comment]
> **What feature should we improve? Please describe.** Localization is the process of adapting content to support different languages and cultural contexts. Currently, our docs are only in English, which limits accessibility for non-English speakers. Localizing our content demonstrates our commitment to accessibility and inclusivity for the hackathon.
> 
> **Describe the solution you'd like** I would like to start by adding French localization to the documentation site. This includes integrating a localization features within the Fumadocs framework, providing a language switcher, and ensuring all content and UI elements are translated into French. The feature should auto-detect what the user's system preference is for language, and use the corresponding language on initial load.
> 
> **Describe alternatives you've considered** I've considered using automatic translation services like Google Translate, but this may not provide accurate or culturally relevant translations. I've also considered localizing only key sections, but this could lead to inconsistencies.
> 
> **Additional context** Implementing localization will require collaboration with native French speakers in our team to ensure accurate translations. Visibility test suites also need to re-architected to account for localized content.

just curious, would this be applicable for all platforms ideally?

**Comment by MFarabi619** [comment]
> > **What feature should we improve? Please describe.** Localization is the process of adapting content to support different languages and cultural contexts. Currently, our docs are only in English, which limits accessibility for non-English speakers. Localizing our content demonstrates our commitment to accessibility and inclusivity for the hackathon.
> > **Describe the solution you'd like** I would like to start by adding French localization to the documentation site. This includes integrating a localization features within the Fumadocs framework, providing a language switcher, and ensuring all content and UI elements are translated into French. The feature should auto-detect what the user's system preference is for language, and use the corresponding language on initial load.
> > **Describe alternatives you've considered** I've considered using automatic translation services like Google Translate, but this may not provide accurate or culturally relevant translations. I've also considered localizing only key sections, but this could lead to inconsistencies.
> > **Additional context** Implementing localization will require collaboration with native French speakers in our team to ensure accurate translations. Visibility test suites also need to re-architected to account for localized content.
> 
> just curious, would this be applicable for all platforms ideally?

Yes, all platforms ideally. The first use case to trial run the implementation and discover best practices at the time that this issue was created was the docs site.
